server-phoenix/lib/helios_web/schema/resolvers/widget.ex

```diff
@@ -1,13 +1,13 @@
 defmodule HeliosWeb.Schema.Resolvers.Widget do
-  alias HeliosWeb.TestData.Weather, as: WeatherData
+  alias HeliosWeb.Clients.WeatherClient
   alias HeliosWeb.TestData.TrafficCams, as: TrafficCamsData
 
   def day_announcements(_parent, _args, _info) do
     {:ok, []}
   end
 
-  def weather(_parent, _args, _info) do
-    {:ok, Jason.decode!(WeatherData.weather)}
+  def weather(parent, _args, _info) do
```

Comment:
I'm wondering if we should be specifying guards here to hint at the parent type.

Reply:
`parent` comes in as a map (of location in this case), so I'm not sure if it's necessary.

Reply:
Won't parent be whatever the resolver sets it? `parent` being `Location` doesn't seem obvious in the widget resolver.

Reply:
Ohhh... I'm sorry, I was reading this as the Location weather resolver. We loaded location onto each widget (in test data) which is how the resolver above can access the latitude and longitude of the location. 

Back to your original question, do you mean checking if parent has a `location` key?